scheduler/pkg/agent/rproxy_grpc.go

```diff
@@ -260,6 +268,104 @@ func (rp *reverseGRPCProxy) ModelInfer(ctx context.Context, r *v2.ModelInferRequ
 	return resp, err
 }
 
+func (rp *reverseGRPCProxy) ModelStreamInfer(stream v2.GRPCInferenceService_ModelStreamInferServer) error {
+	ctx := stream.Context()
+	logger := rp.logger.WithField("func", "ModelStreamInfer")
+	internalModelName, externalModelName, err := rp.extractModelNamesFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	// handle scaling metrics
+	rp.syncScalingMetrics(internalModelName, logger)
+
+	startTime := time.Now()
+	// TODO: check the model is still loaded while the stream is going, not just at the start of the stream
+	err = rp.ensureLoadModel(internalModelName)
+	if err != nil {
+		elapsedTime := time.Since(startTime).Seconds()
+		go rp.metrics.AddModelInferMetrics(externalModelName, internalModelName, metrics.MethodTypeGrpc, elapsedTime, codes.NotFound.String())
+		return status.Error(codes.NotFound, fmt.Sprintf("Model %s not found (err: %s)", internalModelName, err))
+	}
+
+	// Create an outgoing context for the proxy call to service from incoming context
+	outgoingCtx, requestId := rp.createOutgoingCtxWithRequestId(ctx)
+
+	var trailer metadata.MD
+	opts := append(rp.callOptions, grpc.Trailer(&trailer), grpc_retry.Disable())
+
+	client_stream, err := rp.getV2GRPCClient().ModelStreamInfer(outgoingCtx, opts...)
+	if err != nil {
+		logger.WithError(err).Error("Failed to create stream")
+		return err
+	}
+
+	// receive incoming request from envoy and forward them to the model
+	var req_err error
+	done_req := make(chan bool)
```

Comment:
nit: alternatively this could be a wait group? or empty struct as I dont think we make use of the actual boolean value?